crates/bin/pcli/src/command/tx.rs

```diff
@@ -356,27 +348,39 @@ impl TxCmd {
             TxCmd::Swap {
                 input,
                 into,
-                fee,
                 source,
             } => {
                 let input = input.parse::<Value>()?;
                 let into = asset::REGISTRY.parse_unit(into.as_str()).base();
 
-                // Since the swap command consists of two transactions (the swap and the swap claim),
-                // the fee is split equally over both for now.
-                let swap_fee = Fee::from_staking_token_amount((fee / 2).into());
-                let swap_claim_fee = Fee::from_staking_token_amount((fee / 2).into());
-
                 let fvk = app.fvk.clone();
 
                 // If a source address was specified, use it for the swap, otherwise,
                 // use the default address.
                 let (claim_address, _dtk_d) =
                     fvk.incoming().payment_address(AddressIndex::new(*source));
 
+                let params = app
+                    .view
+                    .as_mut()
+                    .context("view service must be initialized")?
+                    .app_params()
+                    .await?;
+
+                let gas_prices = params.fee_params.gas_prices;
+
                 let mut planner = Planner::new(OsRng);
-                planner.fee(swap_fee);
-                planner.swap(input, into.id(), swap_claim_fee.clone(), claim_address)?;
+                planner.set_gas_prices(gas_prices.clone());
+                // The swap claim requires a pre-paid fee, however gas costs might change in the meantime.
+                // This shouldn't be an issue, since the planner will account for the difference and add additional
+                // spends alongside the swap claim transaction as necessary.
+                //
+                // Since the swap claim fee needs to be passed in to the planner to build the swap (it is
+                // part of the `SwapPlaintext`), we can't use the planner to estimate the fee and call the
+                // helper method directly.
```

Comment:
I think a gas adjustment mechanism like the Cosmos SDK uses might also help here. On SDK chains it's common to estimate the gas prices client side, then multiply the estimate by a fudge factor (I've seen 1.3, commonly) to account for the possibility that the estimate is wrong. In our case, we have perfect estimation, but possibly evolving fees.

We could multiply the estimated fee by a gas adjustment, to insulate against the possibility that the prices change.